SEED-platform/seed
# Return inventory records with only the columns specified

#### Any background context you want to provide?
- Previously we had an option to request only specific columns be returned from the backend to improve performance and reduce data transfer.  This PR restores that functionality.
#### What's this PR do?
- Passes an optional profile id to the inventory filter API to return data for the columns in that list settings profile
- Fetches all columns for only selected rows when opening the merge modal
- Added an API resource to return only columns that can be mapped
- Removed deprecated code that prepended `tax_` and `property_` to field names
- Commented unused `calculated_taxlot_ids` and `primary` fields
#### How should this be manually tested?
- On an org with no saved list settings, check that the data returned includes all columns
- Create a list setting (like with `Show only populated columns` and check that only those specific columns are returned.
- Try merging some records - check that only the specific selected records were loaded with all columns in the API response.
